doc/check.texi

```diff
@@ -1314,8 +1316,70 @@ two environment variables available that offers this ability,
 the name of the suite and/or test case you want to run. These
 environment variables can also be a good integration tool for
 running specific tests from within another tool, e.g. an IDE.
-  
-@node Testing Signal Handling and Exit Values, Looping Tests, Selective Running of Tests, Advanced Features
+
+@node Selecting Tests Based on Arbitrary Tags, Testing Signal Handling and Exit Values, Selective Running of Tests, Advanced Features
+@section Selecting Tests Based on Arbitrary Tags
+
+@vindex CK_INCLUDE_TAGS
+@vindex CK_EXCLUDE_TAGS
+
+It can also be useful to be able to dynamically select or exclude
```

Comment:
It can be useful to dynamically include or exclude groups of tests to be run based on criteria other than the suite or test case name. For example, one or more tags can be assigned to test cases. The tags could indicate if a test runs for a long time, so such tests could be excluded in order to run quicker tests for a sanity check. Alternately, tags may be used to indicate which functional areas test cover. Tests can then be run that include all test cases for a given set of functional areas.

In Check, a tag is a string of characters without white space. One or more tags can be assigned to a test case by using the @code{tcase_set_tags} function. This function accepts a string, and multiple tags can be specified by delimiting them with spaces. For example:
```
Suite *s;
TCase *red, *purple;
s = suite_create("Tag Example");

red = tcase_create("Red");
/* Add the 'Red' tag to the test case */
tcase_set_tags(red, "Red");
suite_add_tcase (s, red);
tcase_add_test(red, red_test1);

purple = tcase_create("Purple");
/* Add both 'Red' and 'Blue' tags to test case */
tcase_set_tags(purple, "Red Blue");
suite_add_tcase (s, purple);
tcase_add_test(purple, purple_test1);
```
Once test cases are tagged they may be selectively run in one of two ways:

a) Using Environment Variables
There are two environment variables available for selecting test cases based on tags: CK_INCLUDE_TAGS and CK_EXCLUDE_TAGS. These can be set to a space separated list of tag names. If CK_INCLUDE_TAGS is set then test cases which include at least one tag in common with CK_INCLUDE_TAGS will be run. If CK_EXCLUDE_TAGS is set then test cases with one tag in common with CK_EXCLUDE_TAGS will not be run. In cases where both CK_INCLUDE_TAGS and CK_EXCLUDE_TAGS match a tag for a test case the test will be excluded.

Both CK_INCLUDE_TAGS and CK_EXCLUDE_TAGS can be specified in conjunction with CK_RUN_SUITE or even CK_RUN_CASE in which case they will have the effect of further narrowing the selection.

b) Programmatically
The @code{srunner_run_tagged} function allows one to specify which tags to run or exclude from a suite runner. This can be used to programmatically control which test cases may run.